readthedocs/projects/signals.py

```diff
@@ -11,3 +23,34 @@
 
 # Used to purge files from the CDN
 files_changed = django.dispatch.Signal()
+
+
+@receiver(post_save, sender=Version)
+def enable_addons_on_new_mkdocs_projects(instance, *args, **kwargs):
+    """
+    Enable Addons on projects created after 2024-04-01.
+
+    We removed all the `mkdocs.yml` manipulation that set the `readthedocs` if
+    undefined and injects JS and CSS files to show the old flyout.
+
+    Now, we are enabling addons by default on this projects to move forward
+    with this idea of removing the magic executed on behalves the users and
+    promote addons more.
+
+    Reference https://github.com/readthedocs/addons/issues/72#issuecomment-1926647293
+    """
+    project = instance.project
+
+    if (
+        project.pub_date
+        > datetime.datetime(2023, 4, 1, 0, 0, 0, tzinfo=datetime.timezone.utc)
+        and instance.documentation_type == MKDOCS
+    ):
```

Comment:
Shouldn't this logic always run? We are removing the old injection logic completely in this PR, if an old project is build, they won't have the old injections nor the new addons.

Reply:
We have only a few projects on commercial using this feature only, and we plan to contact them telling how to migrate their configuration. Basically, they just need to define `theme.name: readthedocs` in their `mkdocs.yml` file.

I used this date here as an example, but we should give them more time than 20 days 😄  to migrate. We will contact them via email and write a blog post about this as well before merging this PR.

Reply:
I re-read your comment and I think you are right 👍🏼 . I updated this code to remove the datetime limitation.